src/runtime/hexagon_dma_pool.cpp

```diff
@@ -0,0 +1,176 @@
+#include "HalideRuntime.h"
+#include "mini_hexagon_dma.h"
+#include "scoped_mutex_lock.h"
+#include "hexagon_dma_pool.h"
+
+namespace Halide { namespace Runtime { namespace Internal { namespace HexagonDma {
+
+#define MAX_NUMBER_OF_DMA_ENGINES 8
+#define MAX_NUMBER_OF_WORK_UNITS 4
+
+typedef struct {
+    bool in_use;
+    uint8_t num_of_engines;
+    uint8_t mapped_engines[MAX_NUMBER_OF_WORK_UNITS];
+} hexagon_dma_virtual_engine_t;
+
+typedef struct {
+    bool used;
```

Comment:
Comment, particularly on the difference between used vs. assigned

Reply:
Added